src/commands/link/index.ts

```diff
@@ -109,6 +111,38 @@ export default class LinkIndex extends Command {
       throw new Error(chalk.red("No .git found in this directory. Please initialize a git repository with `git init`."));
     }
 
+    // Check if the current branch is 'main'
+    let currentBranch = "";
+    try {
+      currentBranch = execSync("git symbolic-ref --short HEAD", { encoding: "utf-8" }).trim();
+    } catch (error) {
+      this.log(chalk.red("Unable to determine the current branch. Are you inside a Git repository?"));
+      throw error;
+    }
+
+    if (currentBranch !== "main") {
+      this.log(chalk.red("You must be on the 'main' branch to link your repository."));
+      this.log("Please switch to the 'main' branch:");
+      this.log("  > git checkout main");
+      this.log("or rename your current branch to 'main'.");
+      this.log("  > git branch -m main");
+      this.exit(1);
+    }
+
+    const hasRemoteOrigin = await hasGitRemoteUrl(gitConfigFilePath);
```

Comment:
this hasGitRemoteUrl doesnt need to exist, you're checking if it exists, giving them some logs, and then checking again if it exists in getGitRemoteURL. Move the logs to the getGitRemoteURL, since theres already some there. Or make getGitRemoteURL return the string & an error, and log here

Reply:
i made `getGitRemoteUrl` return `string | null`. It shouldn't dictate what errors to print. That should be the main program's responsibility IMO.